test/parallel/test-http-server-de-chunked-trailer.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that when a status code of 304 is set a Trailer header
+// cannot not be used.
+
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer(common.mustCall(function(req, res) {
+  res.setHeader('Trailer', 'baz');
+  assert.throws(() => res.writeHead(200, {'Content-Length': '24'}),
+                common.expectsError({ code: 'ERR_HTTP_TRAILER_INVALID',
```

Comment:
Perhaps move the object literal into a separate variable and pass that to `common.expectsError()` to make this more readable?:

```js
const trailerInvalidErr = {
  code: 'ERR_HTTP_TRAILER_INVALID',
  message: 'Trailers are invalid with this transfer encoding',
  type: Error
};
assert.throws(() => res.writeHead(200, {'Content-Length': '24'}),
              common.expectsError(trailerInvalidErr));
```